File: accidentes/FicheroAccidentes-Java-IvanRC/src/main/java/mapper/accidente/AccidenteMapper.java
package mapper.accidente;

import dto.accidente.AccidenteDto;
import dto.accidente.ListaAccidentesDto;
import mapper.distrito.DistritoMapper;
import mapper.lesividad.LesividadMapper;
import model.Accidente;
import utils.Utils;

import java.util.List;

public class AccidenteMapper {

    public static AccidenteDto toDto(Accidente accidente){
        return new AccidenteDto(
                accidente.getNumeroExpediente(),
                accidente.getFecha().toString(),
                accidente.getHora().toString(),
                accidente.getLocalizacion(),
                Utils.getValorNumericoToString(accidente.getNumeroCalle()),
                DistritoMapper.toDto(accidente.getDistrito()),
                accidente.getTipoAccidente(),
                accidente.getEstadoMeteorologico(),
                accidente.getTipoDeVehiculo(),
                accidente.getTipoDePersona(),
                accidente.getRangoEdad(),
                accidente.getSexo(),
                LesividadMapper.toDto(accidente.getLesividad()),
                Utils.getValorNumericoToString(accidente.getLongitud()),
                Utils.getValorNumericoToString(accidente.getAltitud()),
                Utils.getPositivoToString(accidente.getEsPositivaEnAlchol(), "S", "N"),
                Utils.getPositivoToString(accidente.getEsPositivaEnDrogas(), "1", "NULL")
    );
    }

    public static Accidente toAccidente(AccidenteDto accidenteDto){
        return new Accidente(
                accidenteDto.getNumeroExpediente(),
                Utils.toLocalDateDDMMYYYY(accidenteDto.getFecha()),
                Utils.toLocalTime(accidenteDto.getHora()),
                accidenteDto.getLocalizacion(),
                Utils.valorNumericoToInt(accidenteDto.getNumeroCalle()),
                DistritoMapper.toDistrito(accidenteDto.getDistrito()),
                accidenteDto.getTipoAccidente(),
                accidenteDto.getEstadoMeteorologico(),
                accidenteDto.getTipoDeVehiculo(),
                accidenteDto.getTipoDePersona(),
                accidenteDto.getRangoEdad(),
                accidenteDto.getSexo(),
                LesividadMapper.toLesividad(accidenteDto.getLesividad()),
                Utils.valorNumericoToDouble(accidenteDto.getLongitud()),
                Utils.valorNumericoToDouble(accidenteDto.getAltitud()),
                accidenteDto.getEsPositivaEnAlchol().equals("S"),
                accidenteDto.getEsPositivaEnDrogas().equals("1")
        );
    }

    public static List<Accidente> toAccidentes(ListaAccidentesDto lista){
        return lista.getAccidentesDto().stream().map(a -> toAccidente(a)).toList();
    }

    public static ListaAccidentesDto toAccidentesDto(List<Accidente> lista){
        return new ListaAccidentesDto(
                lista.stream().map(AccidenteMapper::toDto).toList()
        );
    }
}

File: accidentes/FicheroAccidentes-Kotlin-IvanRC/src/main/kotlin/mapper/accidente/AccidenteMapper.kt
package mapper.accidente

import dto.accidente.AccidenteDto
import dto.accidente.ListaAccidentesDto
import mapper.distrito.toDistrito
import mapper.distrito.toDto
import mapper.lesividad.toDto
import mapper.lesividad.toLesividad
import model.Accidente
import utils.`toLocalDateDD-MM-YYYY`
import utils.toLocalTime

fun Accidente.toDto(): AccidenteDto{
    return AccidenteDto(
        numeroExpediente = this.numeroExpediente,
        fecha = this.fecha.toString(),
        hora = this.hora.toString(),
        localizacion = this.localizacion,
        numeroCalle = if(this.numeroCalle != null) this.numeroCalle.toString() else "NULL",
        distrito = this.distrito.toDto(),
        tipoAccidente = this.tipoAccidente,
        estadoMeteorologico = this.estadoMeteorologico,
        tipoDeVehiculo = this.tipoDeVehiculo,
        tipoDePersona = this.tipoDePersona,
        rangoEdad = this.rangoEdad,
        sexo = this.sexo,
        lesividad = this.lesividad.toDto(),
        longitud = if(this.longitud != null) this.longitud.toString() else "NULL",
        altitud = if(this.altitud != null) this.altitud.toString() else "NULL",
        esPositivaEnAlchol = if(this.esPositivaEnAlchol) "S" else "N",
        esPositivaEnDrogas = if(this.esPositivaEnDrogas) "1" else "NULL"
    )
}

fun AccidenteDto.toAccidente(): Accidente{
    return Accidente(
        numeroExpediente = this.numeroExpediente,
        fecha = `toLocalDateDD-MM-YYYY`(this.fecha),
        hora = toLocalTime(this.hora),
        localizacion = this.localizacion,
        numeroCalle = if(this.numeroCalle.matches(Regex("[0-9]*"))) this.numeroCalle.toInt() else null,
        distrito = this.distrito.toDistrito(),
        tipoAccidente = this.tipoAccidente,
        estadoMeteorologico = this.estadoMeteorologico,
        tipoDeVehiculo = this.tipoDeVehiculo,
        tipoDePersona = this.tipoDePersona,
        rangoEdad = this.rangoEdad,
        sexo = this.sexo,
        lesividad = this.lesividad.toLesividad(),
        longitud = if(this.longitud.matches(Regex("[0-9]+.[0-9]+")))this.longitud.replace(",", ".").replace("\"", "").toDouble() else null,
        altitud = if(this.altitud.matches(Regex("[0-9]+.[0-9]+")))this.altitud.replace(",", ".").replace("\"", "").toDouble() else null,
        esPositivaEnAlchol = this.esPositivaEnAlchol == "S",
        esPositivaEnDrogas = this.esPositivaEnDrogas == "1"
    )
}

fun ListaAccidentesDto.toAccidentes(): List<Accidente>{
    return this.accidentesDto.map { it.toAccidente() }
}

fun List<Accidente>.toAccidentesDto(): ListaAccidentesDto{
    return ListaAccidentesDto(
        this.map { it.toDto() }
    )
}
